src/components/Search/SelectedRevisionItem.tsx

```diff
@@ -82,14 +81,33 @@ function SelectedRevisionItem({
   }, [selectedRevisions]);
 
   const handleClose = () => {
-    if (view == 'search') {
-      handleRemoveRevision(item);
+    if (view == searchView || !editBtnVisible) {
+      removeCheckedRevision(item);
     } else {
       deleteSelectedRevisions(item);
     }
     return;
   };
 
+  const renderCloseIcon = () => {
+    const isSearchViewOrEditBtnVisible = view === searchView || !editBtnVisible;
+    const isBaseAndEditBtnVisible = searchType === 'base' || !editBtnVisible;
```

Comment:
It could be good to have some comment outlining a bit more the various contexts that this component is used in, because there are a lot of cases here.